Lib/gzip.py

```diff
@@ -245,8 +245,17 @@ def __init__(self, filename=None, mode=None,
 
         self.fileobj = fileobj
 
-        if self.mode == WRITE:
-            self._write_gzip_header(compresslevel)
+        try:
+            if self.mode == WRITE:
+                self._write_gzip_header(compresslevel)
+        except BaseException:
```

Comment:
IMO a bare except is fine here:

```suggestion
        except:
```